Documentation/concepts/kubernetes/compatibility.rst

```diff
@@ -23,4 +21,101 @@ All Kubernetes versions listed are compatible with Cilium:
 | 1.12, 1.13, 1.14, 1.15, 1.16, 1.17, 1.18 | * `networking.k8s.io/v1`_ | `CustomResourceDefinition` |
 +------------------------------------------+---------------------------+----------------------------+
 
+Cilium CRD schema validation
+============================
+
+Cilium uses a CRD for its Network Policies in Kubernetes. This CRD might have
+changes in its schema validation, which allows it to verify the correctness of
+a Cilium Clusterwide Network Policy (CCNP) or a Cilium Network Policy (CNP).
+
+The CRD itself has an annotation, ``io.cilium.k8s.crd.schema.version``, with the
+schema definition version. By default, Cilium automatically updates the CRD, and
+its validation, with a newer one.
+
+The following table lists all Cilium Versions and their expected schema
+validation version:
+
++-----------------+----------------+
+| Cilium          | CNP and CCNP   |
+| Version         | Schema Version |
++-----------------+----------------+
+| 1.6.0-rc1       | 1.14           |
++-----------------+----------------+
+| 1.6.0-rc2       | 1.14           |
++-----------------+----------------+
+| 1.6.0-rc3       | 1.14           |
```

Comment:
It'd be nice if we can come up with a workflow that doesn't require us to update this doc as well whenever we create a new release. Could we group these by version instead, perhaps with a link to the relevant variable in the code to `pkg.go.dev` in case we forget to update?

But we shouldn't forget to update if we put clear instructions above the schema version definition to outline the expected steps, including "if you change the schema version, update this ____ docs page".

Reply:
From out-of-band discussion and the item below, it makes sense to increment the version for each release to provide a number space for potential bugfix bumps to the schema, like the one that provoked this PR.

We discussed that this table can be auto-generated to a separate file then imported directly into this page. This will resolve my concern around adding manual steps to the release process by automating it.